### ISSUE TITLE
`ArbField`, `AcbField`, `RealField`, `ComplexField` don't implement `isapprox`

### ISSUE THREAD

**Issue by lgoettgens** [Nemocas/Nemo.jl#1877]
This is required by the AbstractAlgebra ring interface for inexact rings (see https://nemocas.github.io/AbstractAlgebra.jl/dev/ring_interface/#Approximation-(floating-point-and-ball-arithmetic-only))

**Comment by fingolfin** [comment]
Hehe I run into the same tonight (didn't see your issue here, sadly).

They also are missing `hash` method.

I now made PR #1886 as a draft to share my partial work on conformance test for these (my guess is that you also have something similar, so let's collaborate instead of duplicating effort :-)

**Comment by lgoettgens** [comment]
> (my guess is that you also have something similar, so let's collaborate instead of duplicating effort :-)

no, nothing on my side yet. This originated in trying to find a non-exact ring to test the ring interface changes with, in particular where I need `equality` and where `==`.
